healthcareai/trained_models/trained_supervised_model.py

```diff
@@ -183,17 +188,37 @@ def prepare_and_subset(self, dataframe):
             pandas.core.frame.DataFrame: A dataframe that has been run through the pipeline and subsetted to only the columns the model expects.
         """
 
+        # Add response column if not included in prediction data
```

Comment:
I changed the comment to explain the reasoning behind adding a prediction column full of NaNs  and added a condition so that the if statement only triggers if the prediction column is found in colnames_pre_pipeline.

Let me know if you'd still like me to make this into its own function.